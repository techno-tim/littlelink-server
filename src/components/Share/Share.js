import React, { memo } from 'react';
import { string } from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

function Share(props) {
  const { url, title, text } = props;

  const handleSharing = async () => {
    if (navigator.share) {
      await navigator.share({
        url,
        title,
        text,
      });
    }
  };

  return (
    <>
      <a className={'button'} rel="noopener noreferrer" onClick={handleSharing}>
        <FontAwesomeIcon className="icon" icon="fas fa-share-nodes" />
      </a>
    </>
  );
}

export default memo(Share);

Share.propType = {
  url: string.isRequired,
  title: string.isRequired,
  text: string.isRequired,
};
