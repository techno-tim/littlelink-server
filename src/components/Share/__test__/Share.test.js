import Share from '../Share';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

describe('<Share />', () => {
  test('renders without exploding', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Share />, div);
  });
  test('<Share /> snapshot', () => {
    const tree = renderer.create(<Share />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
