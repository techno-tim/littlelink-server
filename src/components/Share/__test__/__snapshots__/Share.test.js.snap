// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Share /> <Share /> snapshot 1`] = `
<a
  className="button"
  onClick={[Function]}
  rel="noopener noreferrer"
/>
`;
